File: src/AZ.Integrator.Client/src/app/features/allegro-orders/models/commands/create-inpost-shipment.command.ts
export interface CreateInpostShipmentCommand {
  sender: Sender;
  receiver: Receiver;
  parcels: Parcel[];
  insurance: Insurance;
  cod: Cod;
  reference: string;
  comments: string;
  externalCustomerId: string;
}

export interface Address {
  street: string;
  buildingNumber: string;
  city: string;
  postCode: string;
  countryCode: string;
}

export interface Sender {
  name: string;
  companyName: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  address: Address;
}

export interface Receiver {
  name: string;
  companyName: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  address: Address;
}

export interface Dimensions {
  length: string;
  width: string;
  height: string;
  unit: string;
}

export interface Weight {
  amount: string;
  unit: string;
}

export interface Parcel {
  id: string;
  dimensions: Dimensions;
  weight: Weight;
  isNonStandard: boolean;
}

export interface Insurance {
  amount: number;
  currency: string;
}

export interface Cod {
  amount: number;
  currency: string;
}

File: src/AZ.Integrator.Application/UseCases/Shipments/Commands/CreateInpostShipment/CreateInpostShipmentCommand.cs
using AZ.Integrator.Application.Common.ExternalServices.ShipX.Models;
using Mediator;

namespace AZ.Integrator.Application.UseCases.Shipments.Commands.CreateInpostShipment;

public record CreateInpostShipmentCommand(
    Sender Sender,
    Receiver Receiver,
    List<Parcel> Parcels,
    Insurance Insurance,
    Cod Cod,
    string Reference,
    string Comments,
    string ExternalCustomerId) : ICommand<ShipmentResponse>;